fair/src/autogluon/fair/utils/group_metrics.py

```diff
@@ -0,0 +1,240 @@
+from distutils.log import error
+
+import numpy as np
+from autogluon.core.metrics import roc_auc
+
+
+class BaseGroupMetric:
+    """building block for GroupMetrics. It does book keeping allowing group metrics to take as raw
+    input either a single array containing TP,FP,FN,TN values broadcast over groups and many
+    different thresholds, or singular vectors corresponding to y_true, y_pred, and groups.
+    Also contains additional annotations: name, and greater_is_better
+    """
+
+    def __init__(self, func: callable, name: str, greater_is_better: bool):
+        self.func = func
+        self.name = name
+        self.greater_is_better = greater_is_better
+
+    def build_array(self, args):
```

Comment:
It is internal logic only used by children of this class. I've updated the doc string.
I'm happy to move it out, but I thought this would make the usage clearer.